src/cleanfigure.m

```diff
@@ -367,6 +368,259 @@ function coarsenLine(meta, handle, minimumPointsDistance)
   return;
 end
 % =========================================================================
+function simplifyLine(meta, handle, targetResolution)
+  % Reduce the number of data points in the line handle by removing
+  % points which add features with area smaller than 1/4 of a pixel at the
+  % target resolution. 
+  %
+  % targetResolution is in format 'WxH@Res', eg. '15x9@600' for a 15cm by 9cm
+  % figure at 600 ppcm.
+  % 
+  % Use targetResolution = 'off' to disable line simplification
+
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  zData = get(handle, 'ZData');
+
+  if ~isempty(zData)
+    % Don't simplify 3d plots
+    return;
+  end
+  if isempty(xData) || isempty(yData)
+      return;
+  end
+  if numel(xData) <= 2
+      return;
+  end
+  if strcmpi(targetResolution,'off')
+      return
+  end
+  [parms,nparms] = sscanf(targetResolution,'%fx%f@%f');
+  
+  if nparms < 3
+      parms(3) = 600;
+  end
+
+  % Get info about log scaling.
+  isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+  vxData = xData;
+  if isXlog
+	  vxData = log10(xData);
+  end
+
+  isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+  vyData = yData;
+  if isYlog
+	  vyData = log10(yData);
+  end
+  
+  % Automatically guess a tol based on the area of the figure and 
+  % the area and resolution of the output
+  a = axis(meta.gca);
+  if ~isXlog
+      xrange = (a(2)-a(1));
+  else
+      xrange = (log10(a(2))-log10(a(1)));
+  end
+  if ~isYlog
+      yrange = (a(4)-a(3));
+  else
+      yrange = (log10(a(4))-log10(a(3)));
+  end
+  tol = xrange*yrange/(4*prod(parms));
+  
+
+  %Split up lines which are seperated by NaNs
+  nans = isnan(vxData) | isnan(vyData);
+  idx = sort([1,find(nans)-1,find(nans)+1,numel(vxData)]);
```

Comment:
@okomarov Thanks, I've added your snippet. Unless anyone has any objections, I'm replacing consecutive of NaNs with just one NaN.